### PR TITLE
refactor handout-tips scripts

### PR DESCRIPTION
**Pull request by jimustafa**
- combine all the handout-tips scripts into one
- expected to help with #88

### CODE REVIEW

File: scripts/tips.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+fig = plt.figure(figsize=(2, 2))
+ax = fig.add_axes(rect, xticks=[], yticks=[])
+
+np.random.seed(1)
+X = np.random.randn(1000, 4)
+cmap = plt.get_cmap("Oranges")
+colors = [cmap(i) for i in [0.1, 0.3, 0.5, 0.7]]
+ax.hist(X, 2, density=True, histtype='bar', color=colors)
+
+plt.savefig("../figures/tip-color-range.pdf")
```
Comment (QuLogic) [inline scripts/tips.py:30]: Should use `fig.savefig` for these.

File: scripts/tips.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+ax.xaxis.set_major_locator(mpl.ticker.MultipleLocator(1.0))
+ax.xaxis.set_minor_locator(mpl.ticker.MultipleLocator(0.2))
+ax.xaxis.set_major_formatter(mpl.ticker.ScalarFormatter())
+ax.xaxis.set_minor_formatter(mpl.ticker.ScalarFormatter())
+ax.tick_params(axis='x', which='minor', rotation=0)
+
+for tick in ax.get_xticklabels(which='both'):
+    tick.set_fontname("Roboto Condensed")
+
+plt.tight_layout()
```
Comment (QuLogic) [inline scripts/tips.py:124]: Can do `fig.tight_layout`, I think?

File: scripts/tips.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+ax.bar(x1, y1, color=color2)
+for i in range(len(x1)):
+    plt.annotate("%d%%" % y1[i],  (x1[i], y1[i]), xytext=(0,1),
+                 fontsize="x-small", color=color2,
+                 textcoords="offset points", va="bottom", ha="center")
+
+ax.bar(x2, y2, color=color2, hatch="/")
+for i in range(len(x2)):
+    plt.annotate("%d%%" % y2[i],  (x2[i], y2[i]), xytext=(0,1),
```
Comment (QuLogic) [inline scripts/tips.py:151]: Should use `ax.annotate`

File: scripts/tips.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+for x in np.linspace(0.01, 10*np.pi-0.01, 100):
+    X.extend([x, x, None])
+    Y.extend([0, np.sin(x), None])
+print(X[:10], Y[:10])
+plt.plot(X, Y, "black")
+plt.xticks([]), plt.yticks([])
+plt.xlim(-0.25, 10*np.pi+0.25)
+plt.ylim(-1.5, 1.5)
+plt.tight_layout()
+plt.savefig("../figures/tip-multiline.pdf", dpi=100)
```
Comment (QuLogic) [inline scripts/tips.py:183]: Should all switch away from `plt`.

**Comment by rougier** [comment]
I prefer to have one file / one idea because I find difficult some time to get the code corresponding to an image when there are several examples in the same file (e.g. gallery). Furthermore, what do we gain by having a single file?

**Comment by timhoffm** [comment]
I don't have a strong opinion either way. But I don't see a big difference in discoverability. The main obstacle is finding the appropriate name. Whether I then open [name].py or search for [name].pdf (from the savefig command) in a single file is no big difference.

**Comment by rougier** [comment]
Look for example https://matplotlib.org/2.0.2/examples/statistics/boxplot_demo.html. Source is compact but as a user, if you're interested in a specific subfigure, it takes your some time to identify the code. Having one script / figure make it easier to get the relevant code.

**Comment by jimustafa** [comment]
Admittedly, consolidating the scripts is probably just a matter of taste. There may be some benefit in being able to more easily recognize code patterns and keeping things DRY. Could also be easier to keep a consistent style with a single file. Either way, not a big deal.

File: scripts/tips.py [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    ax.set_xlim(0, 5)
+    ax.set_ylim(0, 1)
+    ax.patch.set_alpha(0.0)
+
+
+fig = plt.figure(figsize=(5, 0.5))
+fig.patch.set_alpha(0.0)
+n = 1
+
+fontsize = 18
```
Comment (QuLogic) [inline scripts/tips.py:111]: Unused?

File: scripts/tips.py [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+ax = plt.subplot(n, 1, 1)
```
Comment (QuLogic) [inline scripts/tips.py:112]: Join `fig`, and `ax`, I think.
```suggestion
fig, ax = plt.subplots(figsize=(5, 0.5))
fig.patch.set_alpha(0.0)

fontsize = 18
```

File: scripts/tips.py [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+ax2.set_xticklabels([])
+ax2.set_yticklabels([])
+
+fig.savefig("../figures/tip-dual-axis.pdf")
+
+
+# font family
+# -----------------------------------------------------------------------------
+# Setup a plot such that only the bottom spine is shown
+def setup(ax):
```
Comment (QuLogic) [inline scripts/tips.py:88]:
```suggestion
def setup(ax):
    """Setup a plot such that only the bottom spine is shown."""
```

File: scripts/tips.py [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+# hatched
+# -----------------------------------------------------------------------------
+cmap = plt.get_cmap("Oranges")
+color1, color2 = cmap(0.3), cmap(0.5)
+
+mpl.rcParams['hatch.color'] = color1
+mpl.rcParams['hatch.linewidth'] = 8
+
+fig = plt.figure(figsize=(2, 2))
+ax = plt.subplot()
```
Comment (QuLogic) [inline scripts/tips.py:137]:
```suggestion
fig, ax = plt.subplots(figsize=(2, 2))
```

File: scripts/tips.py [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+                       path_effects.Normal()])
+fig.savefig("../figures/tip-outline.pdf")
+
+
+# transparency
+# -----------------------------------------------------------------------------
+fig = plt.figure(figsize=(2, 2), dpi=100)
+ax = fig.add_axes(rect)
+n = 500
+np.random.seed(5)
```
Comment (QuLogic) [inline scripts/tips.py:208]:
```suggestion
np.random.seed(123)
```

**Review by QuLogic — COMMENTED** [review]
Mostly minor things.

But also, I see `tip-post-processing.py` was deleted, and I don't see it in the merged script?

**Comment by rougier** [comment]
Again, I don't see the advantage of having a single script. I think it is easier to have well named isolated scripts that make things easier to find for the user.

**Comment by jimustafa** [comment]
> Again, I don't see the advantage of having a single script. I think it is easier to have well named isolated scripts that make things easier to find for the user.

I get it. Was not sure a decision was made on this PR, so I just rebased so that it _could_ be merged. Combining the scripts could help reduce repetition, especially as we add more per-script configuration, as in #127 and #129.

Anyways, I will close for now, and can reopen if opinion changes. Thanks!

**Pull request closed without merging**